client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -59,8 +61,11 @@ const PatternAssembler: Step = ( { navigation, flow } ) => {
 
 	useEffect( () => {
 		// Require to start the flow from the first step
-		if ( ! selectedDesign ) {
+		if ( ! selectedDesign && flow === SITE_SETUP_FLOW ) {
 			goToStep?.( 'goals' );
+		} else if ( selectedTheme === 'blank-canvas-3' && flow === ASSEMBLER_FLOW ) {
+			// User has selected blank-canvas-3 theme from theme showcase and enter assembler flow
+			setSelectedDesign( BLANK_CANVAS_DESIGN as Design );
```

Comment:
I think we can do it when the user lands on the site assembler flow instead of this step. For example, we can call `useEffect( () => { ... }, [] )` in the `useSteps` function (just like other flow does). But maybe there is a better way to do it.

Reply:
I'll make sure to name it `siteAssemblerFlow`, similar to `siteSetupFlow`, does it make sense?

Reply:
Make sense, now the PA step won't have to preselect the design, this will happens on the Site assembler flow